src/pages/articles/software-engineering/singleton/index.md

```diff
@@ -0,0 +1,45 @@
+---
```

Comment:
maybe expand the directory structure to `src/pages/articles/software-engineering/design-patterns/singleton`